ObservatoryHerald/HeraldQueue.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Observatory.Framework;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using System.Linq;
+using NetCoreAudio;
+
+namespace Observatory.Herald
+{
+    class HeraldQueue
+    {
+        private Queue<NotificationArgs> notifications;
+        private bool processing;
+        private string voice;
+        private VoiceAzureCacheManager azureCacheManager;
+        private Player audioPlayer;
+        
+        public HeraldQueue(VoiceAzureCacheManager azureCacheManager)
+        {
+            this.azureCacheManager = azureCacheManager;
+            processing = false;
+            notifications = new();
+            audioPlayer = new();
+        }
+
+
+        internal void Enqueue(NotificationArgs notification, string selectedVoice)
+        {
+            voice = selectedVoice;
```

Comment:
Notifications are processed async yet voice is global state but provided per-notification -- which leads to some potentially incorrect assumptions/coupling. If selectedVoice is intended to be coupled with the notification, then make this part of NotificationArgs. If intended to be global, I'd suggest decoupling it from Enqueue() and either set it via ctor or via an explicit Voice property (set when HeraldNotifier.Settings is changed?) -- or both.

Reply:
Decoupling it from `Enqueue` is probably the way to go here.